### ISSUE TITLE
PPDet训练PCB数据集，YOLO系列无法输出eval结果

### ISSUE THREAD

**Issue by ouxinyu** [PaddlePaddle/PaddleDetection#3014]
在使用PaddleDetection2.0进行PCB数据集训练的时候，使用FasterRCNN模型是正确的，使用YOLOv3和PPYOLO模型训练的时候，eval功能不起作用，不会输出recall和precision，直接使用eval对生成的模型进行验证的时候也是一样的，不会输出召回率和精确率。
结果如下：

- **PPYOLO**
[05/14 23:03:28] ppdet.utils.checkpoint INFO: Finish loading model weights: C:\Users\Administrator/.cache/paddle/weights\ResNet18_vd_pretrained.pdparams
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
[05/14 23:03:54] ppdet.engine INFO: Epoch: [0] [ 0/24] learning_rate: 0.000125 loss_xy: 0.000000 loss_wh: 0.000000 loss_iou: 0.000000 loss_obj: 2466.539551 loss_cls: 0.000000 loss: 2466.539551 eta: 2:06:54 batch_cost: 26.4392 data_cost: 25.7591 ips: 0.9077 images/s
[05/14 23:09:29] ppdet.engine INFO: Epoch: [0] [10/24] learning_rate: 0.000350 loss_xy: 0.000000 loss_wh: 0.000000 loss_iou: 0.000000 loss_obj: 1.626562 loss_cls: 0.000000 loss: 2.469167 eta: 2:32:13 
batch_cost: 33.4978 data_cost: 33.2016 ips: 0.7165 images/s
[05/14 23:15:22] ppdet.engine INFO: Epoch: [0] [20/24] learning_rate: 0.000575 loss_xy: 0.000000 loss_wh: 0.000000 loss_iou: 0.000000 loss_obj: 0.000043 loss_cls: 0.000000 loss: 0.000043 eta: 2:31:49 
batch_cost: 35.2399 data_cost: 34.8799 ips: 0.6810 images/s
[05/14 23:16:55] ppdet.utils.checkpoint INFO: Save checkpoint: D:/Workspace/ExpResults/PCB_ppyolo_r18vd_coco
C:\Users\Administrator\anaconda3\lib\site-packages\win32\lib\pywintypes.py:2: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp, sys, os
loading annotations into memory...
Done (t=0.04s)
creating index...
index created!
[05/14 23:17:03] ppdet.engine INFO: Eval iter: 0
C:\Users\Administrator\anaconda3\lib\site-packages\ppdet\engine\callbacks.py:171: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
  logger.warn("Evaluation results empty, this may be due to " \
[05/14 23:17:14] ppdet.engine WARNING: Evaluation results empty, this may be due to training iterations being too few or not loading the correct weights.
[05/14 23:17:21] ppdet.engine INFO: Epoch: [1] [ 0/24] learning_rate: 0.000665 loss_xy: 0.115875 loss_wh: 0.228548 loss_iou: 0.344624 loss_obj: 2.916828 loss_cls: 0.370267 loss: 4.001377 eta: 2:22:59 
batch_cost: 31.3453 data_cost: 30.9331 ips: 0.7657 images/s

- **FasterRCNN**
- 05/14 23:25:39] ppdet.utils.checkpoint INFO: Save checkpoint: D:/Workspace/ExpResults/PCB_faster_rcnn_r34_fpn_1x_lite_coco
C:\Users\Administrator\anaconda3\lib\site-packages\win32\lib\pywintypes.py:2: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp, sys, os
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
[05/14 23:25:42] ppdet.engine INFO: Eval iter: 0
[05/14 23:25:53] ppdet.metrics.metrics INFO: The bbox result is saved to bbox.json.
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
[05/14 23:25:53] ppdet.metrics.coco_utils INFO: Start evaluate...
Loading and preparing results...
DONE (t=0.02s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.30s).
Accumulating evaluation results...
DONE (t=0.10s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.048
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.188
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.009
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.164
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.051
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.040
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.156
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.162
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.163
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.171
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004
[05/14 23:25:54] ppdet.engine INFO: Total sample number: 100, averge FPS: 8.860917591580263
[05/14 23:25:54] ppdet.engine INFO: Best test bbox ap is 0.048.
[05/14 23:25:57] ppdet.utils.checkpoint INFO: Save checkpoint: D:/Workspace/ExpResults/PCB_faster_rcnn_r34_fpn_1x_lite_coco
[05/14 23:25:58] ppdet.engine INFO: Epoch: [1] [  0/593] learning_rate: 0.002500 loss_rpn_

**Comment by heavengate** [comment]
评估信息不是已经输出了么，前一半`Average Precision`是准确率，后一半`Average Recall`是召回率
```
Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.048
Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.188
Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.009
Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.164
Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.051
Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.040
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.156
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.162
Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.163
Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.171
Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004
```
